tests/interfaces/test_numpyro/test_numpyro_marginal_ll.py

```diff
@@ -0,0 +1,292 @@
+import chex
+import jax.numpy as jnp
+import jax.random as jr
+import numpyro
+import numpyro.distributions as dist
+import pytest
+from numpyro.contrib.tfp import distributions as tfd
+from numpyro.distributions import constraints
+
+from gpjax.gps import Prior
+from gpjax.interfaces.numpyro import (
+    add_constraints,
+    add_priors,
+    numpyro_dict_params,
+    numpyro_marginal_ll,
+)
+from gpjax.kernels import RBF
+from gpjax.likelihoods import Gaussian
+from gpjax.parameters import initialise
+
+seed = 123
+KEY = jr.PRNGKey(seed)
+
+# TODO: test conjugate posterior
+def _get_conjugate_posterior_params() -> dict:
+    kernel = RBF()
+    prior = Prior(kernel=kernel)
+    lik = Gaussian()
+    posterior = prior * lik
+    params = initialise(posterior)
+    return params, posterior
+
+
+def _gen_training_data(n_samples, n_features, n_latents):
+
+    x = jr.normal(key=KEY, shape=(n_samples, n_features))
+    y = jr.normal(key=KEY, shape=(n_samples, n_latents))
+    return x, y
+
+
+def test_numpyro_marginal_ll_params():
+
+    # create sample data
+    X, y = _gen_training_data(10, 10, 2)
+
+    # initialize parameters
+    params, posterior = _get_conjugate_posterior_params()
+
+    # convert to numpyro-style params
+    numpyro_params = numpyro_dict_params(params)
+
+    # initialize numpyro-style GP model
+    npy_model = numpyro_marginal_ll(posterior, numpyro_params)
+
+    # do one forward pass with context
+    with numpyro.handlers.seed(rng_seed=KEY):
+        model_params = numpyro.handlers.trace(npy_model).get_trace(X, y.T)
```

Comment:
This will then become `Dataset` in the `get_trace()` call.

Reply:
Fixed!